cmd/inspect/block.go

```diff
@@ -0,0 +1,267 @@
+							fmt.Printf("[malformed 'evm.ethereum.v0' transaction: %s]\n", err)
+							break
+						}
+
+						fmt.Printf("Kind:      evm.ethereum.v0\n")
+						fmt.Printf("Hash:      %s\n", tx.Tx.Hash())
+						fmt.Printf("Eth hash:  %s\n", ethTx.Hash())
+						fmt.Printf("Chain ID:  %s\n", ethTx.ChainId())
+						fmt.Printf("Nonce:     %d\n", ethTx.Nonce())
+						fmt.Printf("Type:      %d\n", ethTx.Type())
+						fmt.Printf("To:        %s\n", ethTx.To())
+						fmt.Printf("Value:     %s\n", ethTx.Value())
+						fmt.Printf("Gas limit: %d\n", ethTx.Gas())
+						fmt.Printf("Gas price: %s\n", ethTx.GasPrice())
+						fmt.Printf("Data:\n")
+						if len(ethTx.Data()) > 0 {
+							fmt.Printf("  %s\n", base64.StdEncoding.EncodeToString(ethTx.Data()))
+						} else {
+							fmt.Printf("  (none)\n")
+						}
+					default:
+						fmt.Printf("[module-specific transaction encoding scheme: %s]\n", scheme)
+					}
+				} else {
+					// Regular SDK transaction.
+					fmt.Printf("Kind: oasis\n")
+
+					common.PrintTransactionRaw(npa, &tx.Tx)
+				}
+				fmt.Println()
+
+				// Show result.
+				fmt.Printf("=== Result of transaction %d ===\n", txIndex)
+				switch res := tx.Result; {
+				case res.Failed != nil:
+					fmt.Printf("Status:  failed\n")
+					fmt.Printf("Module:  %s\n", res.Failed.Module)
+					fmt.Printf("Code:    %d\n", res.Failed.Code)
+					fmt.Printf("Message: %s\n", res.Failed.Message)
+				case res.Ok != nil:
+					fmt.Printf("Status: ok\n")
+					fmt.Printf("Data:\n")
+					prettyPrintCBOR("  ", res.Ok)
+				case res.Unknown != nil:
+					fmt.Printf("Status: unknown\n")
+					fmt.Printf("Data:\n")
+					prettyPrintCBOR("  ", res.Unknown)
+				default:
+					fmt.Printf("[unsupported result kind]\n")
+				}
+				fmt.Println()
+
+				// Show events.
+				fmt.Printf("=== Events emitted by transaction %d ===\n", txIndex)
+				if numEvents := len(tx.Events); numEvents > 0 {
+					fmt.Printf("Events: %d\n", numEvents)
+					fmt.Println()
+
+					for evIndex, ev := range tx.Events {
+						prettyPrintEvent("  ", evIndex, ev, evDecoders)
+						fmt.Println()
+					}
+				} else {
+					fmt.Println("No events emitted by this transaction.")
+				}
+			}
+		}
+	},
+}
+
+func prettyPrintCBOR(indent string, data []byte) {
+	var body interface{}
+	if err := cbor.Unmarshal(data, &body); err != nil {
+		fmt.Printf("%s%s\n", indent, base64.StdEncoding.EncodeToString(data))
+		return
+	}
+
+	prettyPrintStruct(indent, data, body)
+}
+
+func prettyPrintStruct(indent string, data []byte, body interface{}) {
+	// TODO: Support the pretty printer.
+	pretty, err := yaml.Marshal(body)
+	if err != nil {
+		fmt.Printf("%s%s\n", indent, base64.StdEncoding.EncodeToString(data))
+		return
+	}
+
+	output := string(pretty)
+	output = strings.ReplaceAll(output, "\n", "\n"+indent)
+	output = strings.TrimSpace(output)
+	fmt.Println(indent + output)
+}
+
+func prettyPrintEvent(indent string, evIndex int, ev *types.Event, decoders []client.EventDecoder) {
```

Comment:
Fields like `address`, `topics` and `data` in events are currently printed as a list of integers. Could you fix it to be an ordinary hex-encoded value. That's the standard fallback for EVM tools I think. Is this the pretty printing you were referring to in https://github.com/oasisprotocol/cli/pull/72#issuecomment-1543584935 ?

```
$ oasis inspect block 5653027 3 --network mainnet   --paratime emerald
...
  --- Event 1 ---
  Module: evm
  Code:   1
  Data:
    - address:
      - 240
      - 43
      - 62
      - 67
      - 115
      - 4
      - 137
      - 33
      - 5
      - 153
      - 37
      - 18
      - 83
      - 159
      - 118
      - 148
      - 35
      - 165
      - 21
      - 203
      topics:
      - - 221
        - 242
        - 82
        - 173
        - 27
        - 226
        - 200
        - 155
        - 105
        - 194
        - 176
        - 104
        - 252
...
```

Reply:
This cannot be done without changing the SDK and should be handled as we add pretty printing there.